test/smoke/framework/logging.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"errors"
+	"os"
+	"time"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+
+	"io/ioutil"
+)
+
+const (
+	logFolder = "logs"
+	logSuffix = ".log"
+)
+
+var (
+	monitoredNamespaces = make(map[string]monitoredNamespace)
+)
+
+// StopPodLogCollector waits until all logs are stored on disc
+func StopPodLogCollector(namespace string) error {
+	monitoredNamespace, exists := monitoredNamespaces[namespace]
+	if !exists {
+		return errors.New("namespace is not monitored")
+	}
+	monitoredNamespace.stopMonitoring <- true
+	waitUntilNamespacePodsLogFinish(monitoredNamespace)
+	return nil
+}
+
+func waitUntilNamespacePodsLogFinish(monitoredNamespace monitoredNamespace) {
+	for _, loggingFinished := range monitoredNamespace.pods {
+		<-loggingFinished
```

Comment:
So the logging is finished when the pod is not running anymore and log stream is over right and here we are waiting for that right ?

Reply:
Yes, waiting until all log finishes, as log files are retrireved when pod terminates.

Reply:
What happens if the pod is never deleted ? for example, if we don't delete the namespace ?
(could be useful sometimes)

Reply:
Are chains/channels auto closed on Go?

Reply:
Good point, will address it.

Reply:
> Are chains/channels auto closed on Go?

They aren't, thanks for reminding. Will close the opened channels.